gateway/mw_streaming.go

```diff
@@ -150,20 +190,56 @@ func (s *StreamingMiddleware) Init() {
 
 	s.Logger().Debug("Initializing default stream manager")
 	s.defaultStreamManager = s.createStreamManager(nil)
+
+	// Start garbage collection routine
+	go func() {
+		ticker := time.NewTicker(StreamGCInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				s.garbageCollect()
+			case <-s.ctx.Done():
+				return
+			}
+		}
+	}()
 }
 
 func (s *StreamingMiddleware) createStreamManager(r *http.Request) *StreamManager {
-	newStreamManager := &StreamManager{
-		muxer:  mux.NewRouter(),
-		mw:     s,
-		dryRun: r == nil,
+	streamsConfig := s.getStreamsConfig(r)
+	configJSON, _ := json.Marshal(streamsConfig)
+	cacheKey := fmt.Sprintf("%x", md5.Sum(configJSON))
```

Comment:
100% should be replaced by smth like sha256 or similar. md5 is too weak (I know it was part of orig implementation, but still..)

Reply:
Replaced by sha256.